content/technotes/2020-05-07-rmd-citations/index.Rmd

```diff
@@ -0,0 +1,303 @@
+
+* [knitcitations](https://github.com/cboettig/knitcitations) by [Carl Boettiger](/author/carl-boettiger) allows you to not only use the keys, but also DOIs and URLs to cite a paper. E.g. `citet("10.1098/rspb.2013.1372")` will create a citation to `@Boettiger_2013` after querying the web; and `write.bibtex(file = "references.bib")` will allow you to cache entries in a bibliography file.
+
+Both these packages import [RefManageR](https://docs.ropensci.org/RefManageR/) by [Mathew W. McLean](http://mwmclean.github.io/), a package that _provides tools for importing and working with bibliographic references_, and that has been [peer-reviewed by rOpenSci](https://github.com/ropensci/software-review/issues/119).
+
+### Citation workflow without Pandoc, with RefManageR
+
+Now, for generating Markdown content like this post source, one cannot use the tools presented earlier.
+In our blog guide, we explain that citations are footnotes [whose text is APA formatted](https://blogguide.ropensci.org/technical.html#addcitation) and we explain how to [find that citation text for a package or article](https://blogguide.ropensci.org/technical.html#how-to-find-citation-text-for-a-package-or-article).
+In most cases, authors are only citing a few references, so that is fine.
+However, sometimes authors would like to use an existing .bib file, that they might have e.g. exported from [Zotero](https://www.zotero.org/).
+
+Thankfully, using RefManageR, we can provide a workflow for adding citations from a bib file, by creating functions similar to knitcitations' functions.
+
+In the following examples, we'll use the .bib file below,
+
+```{r, results="asis", echo = FALSE, warning = FALSE}
+glue::glue_collapse(
+  c("```bibtex",
+    readLines("refs.bib"),
+    "```"),
+  sep = "\n")
+```
+
+#### Create a bibliography object
+
+We'll start by creating a `BibEntry` object by calling `RefManageR::ReadBib()`: 
+
+```{r readbib}
+mybib <- RefManageR::ReadBib("refs.bib", check = FALSE) 
+mybib
+class(mybib)
+```
+
+#### Create and use a cite function
+
+Then in another R code chunk, we'll create a function which writes the citations in the format we want. Our citation format is a footnote: `[^key]`.
+
+```{r createf}
+cite <- function(key, bib = mybib) {
+  # create a silent reference so RefManageR
+  # knows we've used this entry
+  RefManageR::NoCite(bib, key)
+  # create the string we need in the Markdown document
+  paste0("[^", key, "]")
+}
+```
+
+From the three entries in the BibTeX file, let us cite two of them:
+
+"what a nice package&#96;r cite("refmanager")&#96; built on top of R&#96;r cite("my-citation-key-for-r")&#96;"
+
+when rendered becomes: 
+
+"what a nice package`r cite("refmanager")` built on top of R`r cite("my-citation-key-for-r")`".
+
```

Comment:
This looks a little funny in the preview, what about something like this (I'm putting this in as actual text, not a suggestion, because with all the code blocks it's really hard to make it look right. The last example would be the same as the second last, just outside of a code block so the footnotes would render correctly:


From the three entries in the BibTeX file, let us cite two of them:

In the Rmd file we write:
```
what a nice package`r cite("refmanager")` built on top of R`r cite("my-citation-key-for-r")`.
```

Which when knit, becomes this in the md file:

```
what a nice package[^refmanager] built on top of R[^my-citation-key-for-r].
```

Which when rendered by Hugo, becomes this in the html file:

what a nice package[^refmanager] built on top of R[^refmanager].

Reply:
right, you mean adding the Markdown result as an intermediary step?

Reply:
Yes, and having the first two steps presented as code blocks, and the final step as regular html text

Reply:
Ok, thanks! I think it looks better now. I could not make the line break before the last step work so I removed it. :sob:

Reply:
Looks great!

Reply:
:relieved: (these were good suggestions but escaping things is hard :sweat_smile: )

Reply:
I enjoyed "escape escaping hell" :grin: